Dockerfile

```diff
@@ -19,13 +21,16 @@ ENV GO111MODULE=on
 RUN go mod download
 RUN go mod verify
 
-# Copy everything from the current directory to the PWD(Present Working Directory) inside the container
-COPY . .
-
-# Build the binary
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build \
-      -ldflags='-w -s -extldflags "-static"' -a \
-       -o /go/bin/goshimmer
+# 1. Mount everything from the current directory to the PWD(Present Working Directory) inside the container
+# 2. Mount the build cache volume
+# 3. Build the binary
+# 4. Verify that goshimmer binary is statically linked
+RUN --mount=target=. \
+    --mount=type=cache,target=/root/.cache/go-build \
+    CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build \
+    -ldflags='-w -s -extldflags "-static"' \
+    -o /go/bin/goshimmer; \
+    ./check_static.sh
```

Comment:
Is the static check really needed? Shouldn't otherwise the build with `-static` fail anyways?

Reply:
We don't need it with 99.99% probability. But since we don't rebuild everything from scratch as with `-a`. I added that check to also be on the safe side :)